sql/mysql/sqlspec_test.go

```diff
@@ -0,0 +1,282 @@
+			},
+		},
+		{
+			spec: specutil.NewCol("int8", "int8"),
+			expected: &schema.IntegerType{
+				T:        tTinyInt,
+				Unsigned: false,
+			},
+		},
+		{
+			spec: specutil.NewCol("int64", "int64"),
+			expected: &schema.IntegerType{
+				T:        tBigInt,
+				Unsigned: false,
+			},
+		},
+		{
+			spec: specutil.NewCol("uint64", "uint64"),
+			expected: &schema.IntegerType{
+				T:        tBigInt,
+				Unsigned: true,
+			},
+		},
+		{
+			spec: specutil.NewCol("string_varchar", "string", specutil.LitAttr("size", "255")),
+			expected: &schema.StringType{
+				T:    tVarchar,
+				Size: 255,
+			},
+		},
+		{
+			spec: specutil.NewCol("string_mediumtext", "string", specutil.LitAttr("size", "100000")),
+			expected: &schema.StringType{
+				T:    tMediumText,
+				Size: 100_000,
+			},
+		},
+		{
+			spec: specutil.NewCol("string_longtext", "string", specutil.LitAttr("size", "17000000")),
+			expected: &schema.StringType{
+				T:    tLongText,
+				Size: 17_000_000,
+			},
+		},
+		{
+			spec: specutil.NewCol("varchar(255)", "varchar(255)"),
+			expected: &schema.StringType{
+				T:    tVarchar,
+				Size: 255,
+			},
+		},
+		{
+			spec: specutil.NewCol("decimal(10, 2) unsigned", "decimal(10, 2) unsigned"),
+			expected: &schema.DecimalType{
+				T:         tDecimal,
+				Scale:     2,
+				Precision: 10,
+			},
+		},
+		{
+			spec: specutil.NewCol("blob", "binary"),
+			expected: &schema.BinaryType{
+				T: tBlob,
+			},
+		},
+		{
+			spec: specutil.NewCol("tinyblob", "binary", specutil.LitAttr("size", "16")),
+			expected: &schema.BinaryType{
+				T:    tTinyBlob,
+				Size: 16,
+			},
+		},
+		{
+			spec: specutil.NewCol("mediumblob", "binary", specutil.LitAttr("size", "100000")),
+			expected: &schema.BinaryType{
+				T:    tMediumBlob,
+				Size: 100_000,
+			},
+		},
+		{
+			spec: specutil.NewCol("longblob", "binary", specutil.LitAttr("size", "20000000")),
+			expected: &schema.BinaryType{
+				T:    tLongBlob,
+				Size: 20_000_000,
+			},
+		},
+		{
+			spec:     specutil.NewCol("enum", "enum", specutil.ListAttr("values", "a", "b", "c")),
+			expected: &schema.EnumType{Values: []string{"a", "b", "c"}},
+		},
+		{
+			spec:     specutil.NewCol("bool", "boolean"),
+			expected: &schema.BoolType{T: "boolean"},
+		},
+		{
+			spec:     specutil.NewCol("decimal", "decimal", specutil.LitAttr("precision", "10"), specutil.LitAttr("scale", "2")),
+			expected: &schema.DecimalType{T: "decimal", Precision: 10, Scale: 2},
+		},
+		{
+			spec:     specutil.NewCol("float", "float", specutil.LitAttr("precision", "10")),
+			expected: &schema.FloatType{T: "float", Precision: 10},
+		},
+		{
+			spec:     specutil.NewCol("float", "float", specutil.LitAttr("precision", "25")),
+			expected: &schema.FloatType{T: "double", Precision: 25},
+		},
+	} {
+		t.Run(tt.spec.Name, func(t *testing.T) {
+			// todo(rotemtam): test uses unexported method, change test after implementing driver Marshal
+			columnType, err := convertColumnType(tt.spec)
```

Comment:
Is there any reason for testing unexported functions?

Reply:
there's are good tests that i want to keep during the refactor
but i need to rewrite the actual test code for them since i made convertColumnType a private func.
there's a bug fixed in #192 that will make the code much nicer so in the meantime this TODO is here and the private method.